whitepapers/0010_ibc_generic_messaging

```diff
@@ -0,0 +1,92 @@
+
+For cosmos chains, we update the core bridge contract to instead send this message over IBC to wormchain. Then a wormchain contract receives the message to emit and actually emits it, which the guardians then pick up.
+
+Specifically, we implement two new cosmwasm smart contracts: `wormhole-ibc` and `wormchain-ibc-receiver`.
+
+The `wormhole-ibc` contract is meant to replace the `wormhole` core bridge contract on cosmos chains. It imports the `wormhole` contract as a library and delegates core functionality to it before and after running custom logic:
+- During execution, it delegates all logic to the core bridge library. After the core bridge library has finished execution, it will send the response as an IBC message to the `wormchain-ibc-receiver` contract <b>if and only if</b> the execution message is of type `wormhole::msg::ExecuteMsg::PostMessage`.
+
+Sending an IBC packet requires choosing an IBC channel to send over. IBC `(channel_id, port_id)` pairs are unique. All IBC-enabled cosmwasm contracts follow a standard `port_id` format: `wasm.<contract_address>`. Therefore, to choose a correct channel the `wormhole-ibc` contract will need to know the address of the `wormchain-ibc-receiver` contract - it will select any channel that is paired with the expected port. 
+
+The `wormchain-ibc-receiver` contract will be deployed on wormchain and is meant to receive the IBC messages that the `wormhole-ibc` contract sends from various cosmos chains. Its only responsibility is to receive the IBC message, send an IBC acknowledgement to the source chain, and then emit the message for the guardian node to observe.
+
+### IBC Relayers
+
+All IBC communication is facilitated by [IBC relayers](https://ibcprotocol.org/relayers/). Since these are lightweight processes that need to only listen to blockchain RPC nodes, each (only several is also acceptable) wormhole guardian can run a relayer.
+
+The guardian IBC relayers are configured to connect the `wormchain-ibc-receiver` contract on wormchain to the various `wormhole-ibc` contracts on the cosmos chains that wormhole supports.
+
+### Guardian Node Watcher
+
+We will modify the cosmos guardian watcher to watch the `wormchain-ibc-receiver` contract on wormchain for the messages from the designated `wormhole-ibc` contracts on supported cosmos chains. This is nearly identical to the current model and can be a drop in replacement.
+
+The new guardian watcher can verify that messages originate from the chain they claim to originate from by checking the IBC connection ID. The `wormchain-ibc-receiver` contract logs the connection ID that the message was received over. Since IBC connections are unique and can never be updated or closed ([docs](https://github.com/cosmos/ibc/tree/main/spec/core/ics-003-connection-semantics#sub-protocols)), the guardians can associate connection IDs with specific chains.
```

Comment:
```suggestion
The new guardian watcher verifies messages originate from the chain they claim to originate from by checking the IBC connection ID. The `wormchain-ibc-receiver` contract logs the connection ID the message was received from. Since IBC connections are unique and can never be updated or closed ([docs](https://github.com/cosmos/ibc/tree/main/spec/core/ics-003-connection-semantics#sub-protocols)), the guardians can associate connection IDs with specific chains.
```